README.md

```diff
@@ -12,194 +12,129 @@ shared. These resources have a different lifecycle than resources in workspaces.
-
-            var vNet =
-                    azureResourceManager
-                            .networks()
-                            .define(definitionContext.resourceNameGenerator().nextName(20))
-                            .withRegion(Region.US_EAST2)
-                            .withExistingResourceGroup(definitionContext.resourceGroup())
-                            .withAddressSpace("10.0.0.0/28")
-                            .withSubnet("compute", "10.0.0.0/29")
-                            .withSubnet("storage", "10.0.0.8/29");
-
-            return definitionContext
-                    .deployment()
-                    .withResourceWithPurpose(storage, ResourcePurpose.SHARED_RESOURCE)
-                    .withVNetWithPurpose(vNet, "compute", SubnetResourcePurpose.WORKSPACE_COMPUTE_SUBNET)
-                    .withVNetWithPurpose(vNet, "storage", SubnetResourcePurpose.WORKSPACE_STORAGE_SUBNET);
-        }
+        
+    @Override
+    public List<ResourceNameRequirements> getResourceNameRequirements() {
+      return List.of(
+              new ResourceNameRequirements(
+                      getResourceType(), ResourceNameGenerator.MAX_VNET_NAME_LENGTH));
     }
 ```
 
-Resources are defined using the standard Azure Java SDK but with the following constraints to consider:
+It is important that step's implementation should be idempotent. Please take a look at Stairway developer guide [here](https://github.com/DataBiosphere/stairway/blob/develop/FLIGHT_DEVELOPER_GUIDE.md).
 
-- The purpose of the resources must be indicated in the deployment.
-- Resources not included in the deployment won't be created.
-- The `create()` method in the resource definition must not be called.
-- The resource definition must have the required configuration for a creation before it can be added to the deployment.
+Resource creation is defined using the standard Azure Java SDK. Together with defining desired properties for a resource it is also important to assign specific tags to a resource based on requirements. 
 
 ### Receiving Parameters
 
-The `DefinitionContext` contains the property `Map<String, String> parameters`, which can be used to receive parameter
-for the definition of a landing zone.
-
-A few points to consider when implementing a definition that requires parameters:
-
-- You must add the necessary validation for these parameters.
-- `parameters` can be  `null`
+Each step has access to `ParameterResolver`. This class allows accessing a specific parameters. All default values for parameters are set in `LandingZoneDefaultParameters`. 
 
 ### Naming Resources and Idempotency
 
-You can use the resource name generator in the deployment context to guarantee that names are consistent in retry
-attempts.
+Each step which creates Azure resource should provide requirements for name generation. For doing this it should provide unique (should be unique across all steps withing a flight) resource type 
```

Comment:
Suggested wording:

Each step which creates an Azure resource should provide requirements for name generation. For doing this, it should provide a unique resource type (unique across all steps within the flight) and also return a list of `ResourceNameRequirements` (in general, a step can create more than one Azure resource). Please take a look at the step example above.